### PR TITLE
initial work on biases for model with implicit feedback

### PR DESCRIPTION
**Pull request by dselivanov**
related to #53 

> INFO  [17:41:35.834] starting factorization with 8 threads 
INFO  [17:41:37.072] iter 1 (items) loss = 3.7734 
INFO  [17:41:37.481] iter 1 (users) loss = 4.0005 
INFO  [17:41:38.686] iter 2 (items) loss = 3.0621 
INFO  [17:41:39.094] iter 2 (users) loss = 3.0296 
INFO  [17:41:40.301] iter 3 (items) loss = 2.8185 
INFO  [17:41:40.701] iter 3 (users) loss = 2.9594 
INFO  [17:41:41.924] iter 4 (items) loss = 2.8183 
INFO  [17:41:42.327] iter 4 (users) loss = 2.9471 
INFO  [17:41:43.584] iter 5 (items) loss = 2.8426 
INFO  [17:41:43.994] iter 5 (users) loss = 2.9721 

- loss doesn't decrease monotonically anymore (strange!)
- on lastfm360k model with biases shows worse map@k compared to the model without biases (which strange as model is more expressive)
- item bias is highly correlated with item popularity (> 0.8 spearman correlation ) (as expected)

cc @david-cortes

### CODE REVIEW

**Comment by david-cortes** [comment]
Thing is, you're subtracting the bias only from the non-missing entries, whereas it needs to be subtracted from all.

Do you see an increase in MAP@K with cmfrec?

**Comment by dselivanov** [comment]
@david-cortes should I use `CMF` or `CMF_implicit` function?

**Comment by david-cortes** [comment]
`CMF`, as `CMF_implicit` doesn't support biases and adds +1 to the weights.

File: R/model_WRMF.R [hunk 1]
```diff
@@ -358,6 +358,10 @@ WRMF = R6::R6Class(
         res = float(0, nrow = private$rank, ncol = nrow(x))
       }
 
+      if (private$with_user_item_bias) {
+        res[1, ] = ifelse(private$precision == "double", 1.0, float::fl(1.0))
```
Comment (david-cortes) [inline R/model_WRMF.R:362]: You might want to change this line: https://github.com/wrathematics/float/issues/36
